File: backend/app/schemas/purchase.py
"""Purchase API schemas."""
from datetime import datetime
from decimal import Decimal
from uuid import UUID
from pydantic import BaseModel, ConfigDict, Field

class PurchaseCreate(BaseModel):
    quantity: int = Field(gt=0)

class PurchaseResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: UUID
    vehicle_id: UUID
    quantity: int
    purchase_price: Decimal
    total_price: Decimal
    created_at: datetime

class PurchaseHistoryResponse(BaseModel):
    purchase_id: UUID
    vehicle_id: UUID
    make: str
    model: str
    category: str
    quantity: int
    purchase_price: Decimal
    total_price: Decimal
    purchase_date: datetime

class PaginationMeta(BaseModel):
    page: int
    page_size: int
    total: int
    total_pages: int

class PaginatedPurchaseResponse(BaseModel):
    items: list[PurchaseHistoryResponse]
    meta: PaginationMeta

class RevenueSummaryResponse(BaseModel):
    total_revenue: float
    total_purchases: int
    total_units_sold: int

File: frontend/src/types/purchase.ts
/** TypeScript types for the Customer Purchase History module. */

export interface Purchase {
  purchase_id: string;
  vehicle_id: string;
  make: string;
  model: string;
  category: string;
  quantity: number;
  purchase_price: number;
  total_price: number;
  purchase_date: string; // ISO-8601 datetime string
}

export interface PaginationMeta {
  page: number;
  page_size: number;
  total: number;
  total_pages: number;
}

export interface PaginatedPurchaseResponse {
  items: Purchase[];
  meta: PaginationMeta;
}

export interface PurchaseHistoryParams {
  page?: number;
  page_size?: number;
  make?: string;
  model?: string;
}
